src/NonlinearArithmetic/Mul.dfy

```diff
@@ -0,0 +1,567 @@
+// RUN: %dafny /compile:0 /noNLarith "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+/***********************************************************************************
+*  Original: Copyright (c) Microsoft Corporation
+*  SPDX-License-Identifier: MIT
+*  
+*  Modifications and Extensions: Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+************************************************************************************/
+
+/* Every lemma comes in 2 forms: 'lemma_property' and 'lemma_property_auto'. The former takes arguments and may 
+be more stable and less reliant on Z3 heuristics. The latter includes automation and its use requires less effort */
+
+include "Internals/MulInternalsNonlinear.dfy"
+include "Internals/MulInternals.dfy"
+
+module Mul {
+
+  import MulINL = MulInternalsNonlinear
+  import opened MulInternals
+
+  /* the common syntax of multiplication results in the same product as multiplication 
```

Comment:
Can you write "builtin" instead of "common"? Took me a second to figure out what you meant.

Reply:
Thanks for the wording suggestion!